### ISSUE TITLE
JetBrains Fleet: Add a 1-click button for launching Fleet

### ISSUE THREAD

**Issue by matifali** [coder/modules#246]
As per [this](https://youtrack.jetbrains.com/issue/FL-23947/Allow-to-connect-to-a-remote-machine-over-SSH-via-a-URL-handler#focus=Comments-27-9783970.0-0) Fleet issue, it will soon support launching remote environments via a URI scheme. 

With Fleet 1.36 (the version after the upcoming version 1.35), Fleet URI handler will support connecting to SSH directly using URIs of this format:

```
fleet://fleet.ssh/<hostname>?<query_params>
```

with possible params being

- `username`
- `port`
- `pwd `(the working directory to open)
- `forceNewHost `(possible values are `true `and `false`, default is `false `- Setting to true will let Fleet ignore previous known hosts and always create a new connection)

`username `and `port `can also be specified alongside the `hostname` in common SSH URI syntax, i.e. `<username>@<hostname>:<port>`. To avoid ambiguities, `pwd` can only be specified explicitly.

Only the `hostname` is required. If some parameters are not specified, Fleet will try to use the closest known match in previous connections or fall back to default values. `.ssh/config` is considered as usual. Not specifying `pwd` will open an empty workspace on the host.

Examples:

```
fleet://fleet.ssh/example.org?username=foo&pwd=/home/foo/my_test_project

fleet://fleet.ssh/foo@example.org?pwd=/home/foo/my_test_project
```

The current process to connect to a Coder workspace is documented in https://github.com/coder/coder/pull/13179

@kylecarbs @code-asher, what are your thoughts on how we can dynamically populate the URI? We have our plugins for VSCode Desktop and JetBrains gateway, which populates the `.ssh/config`.

**Comment by kylecarbs** [comment]
Is it possible to have a Fleet extension?

**Comment by matifali** [comment]
@kylecarbs not yet. But it's on their long-term roadmap. No developers docs ATM.

**Comment by code-asher** [comment]
Yeah without a plugin the best we can do is document that `coder config-ssh` has to be ran manually and then use the `coder.workpsacename` hostname in the URI, I think.

**Comment by kylecarbs** [comment]
This is insane, but I'm pretty sure Chrome handles `ssh://`.

In some magically insane world, we could add a small sniffer to the initial HTTPS packet for the SSH prefix (e.g. `SSH-`), and then authenticate using a secret username. Then we could one-click open anything, even without an extension!

```
ssh mymagicstring--dev--kylecarbs@dev.coder.com
```

**Comment by code-asher** [comment]
Woaaaah that is actually dope.

**Comment by michaelbrewer** [comment]
nice, what awesome about fleet is that it's free, for now.

**Comment by michaelbrewer** [comment]
Maybe in preparation the fleet icon can be added to the coder server ;) 

Also you do need have your ssh config updated ahead of time.

**Comment by michaelbrewer** [comment]
<img width="910" alt="Screenshot 2024-05-23 at 3 36 54 PM" src="https://github.com/coder/modules/assets/5442469/a8334de8-d30a-4ac9-8242-cb2e7906b479">

For sure you can access the server via chrome:  `ssh://coder.platform-onbarding.main` and with fleet it would be `fleet://fleet.ssh/foo@example.org?pwd=/workspace/platform-onboarding`. The template would have to get the remote name ie: `coder.platform-onbarding.main`, for which i assume it `coder.<workspace_id>.`<agent_id>`?

**Comment by matifali** [comment]
> or which i assume it coder.<workspace_id>.<agent_id>

It should be `coder.<workspace_id>.<agent_name>` given that `coder config-ssh` is already run.

**Comment by michaelbrewer** [comment]
i see that fleet 1.37 now out and this url schema works well :)

**Comment by michaelbrewer** [comment]
@matifali i added the fleet icon to coder so long https://github.com/coder/coder/pull/14021

**Comment by michaelbrewer** [comment]
> > or which i assume it coder.<workspace_id>.<agent_id>
> 
> It should be `coder.<workspace_id>.<agent_name>` given that `coder config-ssh` is already run.

And there is no way to know that the coder cli was installed locally or that `coder config-ssh`  was run

**Comment by matifali** [comment]
> And there is no way to know that the coder cli was installed locally or that coder config-ssh was run

Sadly yes for now. It might change in future when Fleet support extensions or We find another way probably through Coder chrome extension.

**Comment by michaelbrewer** [comment]
The same problem exists for people using JetBrains IDE (you need to pre-install Jetbrain Gateway).

**Comment by matifali** [comment]
May be we can use https://www.jetbrains.com/help/fleet/install-on-a-remote-machine.html
